azurerm/resource_arm_data_factory.go

```diff
@@ -0,0 +1,396 @@
+		}
+	}
+
+	d.SetId(*resp.ID)
+
+	return resourceArmDataFactoryRead(d, meta)
+}
+
+func resourceArmDataFactoryRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).dataFactoryClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	name := id.Path["factories"]
+
+	resp, err := client.Get(ctx, resourceGroup, name, "")
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("Error retrieving Data Factory %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	d.Set("name", resp.Name)
+	d.Set("resource_group_name", resourceGroup)
+	if location := resp.Location; location != nil {
+		d.Set("location", azureRMNormalizeLocation(*location))
+	}
+
+	repoType, repo := flattenArmDataFactoryRepoConfiguration(&resp)
+	if repoType == datafactory.TypeFactoryVSTSConfiguration {
+		if err := d.Set("vsts_configuration", repo); err != nil {
+			return fmt.Errorf("Error setting `vsts_configuration`: %+v", err)
+		}
+	} else {
+		d.Set("vsts_configuration", []interface{}{})
+	}
+	if repoType == datafactory.TypeFactoryGitHubConfiguration {
+		if err := d.Set("github_configuration", repo); err != nil {
+			return fmt.Errorf("Error setting `github_configuration`: %+v", err)
+		}
+	} else {
+		d.Set("github_configuration", []interface{}{})
+	}
+	if repoType == datafactory.TypeFactoryRepoConfiguration {
+		d.Set("vsts_configuration", repo)
+		d.Set("github_configuration", repo)
+	}
+
+	if err := d.Set("identity", flattenArmDataFactoryIdentity(resp.Identity)); err != nil {
+		return fmt.Errorf("Error flattening `identity`: %+v", err)
+	}
+
+	flattenAndSetTags(d, resp.Tags)
+
+	return nil
+}
+
+func resourceArmDataFactoryDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).dataFactoryClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	name := id.Path["factories"]
+
+	response, err := client.Delete(ctx, resourceGroup, name)
+	if err != nil {
+		if !utils.ResponseWasNotFound(response) {
+			return fmt.Errorf("Error deleting Data Factory %q (Resource Group %q): %+v", name, resourceGroup, err)
+		}
+	}
+
+	return nil
+}
+
+func expandArmDataFactoryRepoConfiguration(d *schema.ResourceData) (bool, datafactory.BasicFactoryRepoConfiguration) {
+	if vstsList, ok := d.GetOk("vsts_configuration"); ok {
+		vsts := vstsList.([]interface{})[0].(map[string]interface{})
+		accountName := vsts["account_name"].(string)
+		branchName := vsts["branch_name"].(string)
+		projectName := vsts["project_name"].(string)
+		repositoryName := vsts["repository_name"].(string)
+		rootFolder := vsts["root_folder"].(string)
+		tenantID := vsts["tenant_id"].(string)
+		// https://github.com/Azure/go-autorest/issues/307
```

Comment:
_minor_ expanding on this comment as to whats going on beyond a link could be useful

Reply:
Hbuckle found a weird case where you can pass the struct or the address of the struct into the sdk. If you didn't pass the address though, the sdk broke. I don't think we need to track that inside Terraform though since we can just pass the address like we normally do.